witness/tests/Test/Dscp/Witness/Block/BlockSpec.hs

```diff
@@ -0,0 +1,157 @@
+module Test.Dscp.Witness.Block.BlockSpec where
+
+import Control.Lens (ix, to)
+import qualified Data.List as L
+import qualified GHC.Exts as Exts
+import Test.QuickCheck.Modifiers (getPositive)
+import Test.QuickCheck.Monadic (pre)
+
+import Dscp.Core
+import Dscp.Crypto
+import Dscp.Resource.Keys
+import Dscp.Util.Test
+import Dscp.Witness
+import Test.Dscp.Witness.Mode
+
+-- | Submit block to validation.
+submitBlock
+    :: (HasWitnessConfig, WithinWriteSDLock)
+    => Block -> WitnessTestMode ()
+submitBlock = void . applyBlock
+
+-- | Make sound list of blocks.
+-- Given secret key is assumed to correspond to educator - author of blocks -
+-- and witness who publishes this block.
+makeBlocksChain :: HasWitnessConfig => Int -> SecretKey -> NonEmpty Block
+makeBlocksChain n issuer
+    | n <= 0 = error "makeBlocksChain: n <= o"
+    | otherwise =
+        Exts.fromList $
+        fix $ \blocks ->
+        zip3 [1..n] (genesisBlock : blocks) issuingSlots
+        <&> \(i, prevBlock, hSlotId) ->
+            let hDifficulty = fromIntegral i
+                hPrevHash = hash (bHeader prevBlock)
+                bBody = BlockBody []
+                toSign = BlockToSign hDifficulty hSlotId hPrevHash (hash bBody)
+                hSignature = sign issuer toSign
+            in Block
+            { bBody
+            , bHeader = Header
+              { hIssuer = toPublic issuer
+              , hDifficulty, hSlotId, hPrevHash, hSignature
+              }
+            }
+  where
+    issuerAddr = mkAddr $ toPublic issuer
+    issuingSlots = filter (committeeOwnsSlot testCommittee issuerAddr) [1..999]
```

Comment:
I set max bound just because throwing error is better than hanging, for the case when one mixes up with test case and passes `issuer` which owns no slot.
Although I see now that error won't be thrown acutally, fixing it.